src/libcore/cell.rs

```diff
@@ -347,6 +350,68 @@ pub enum BorrowState {
     Unused,
 }
 
+/// An error returned by [`RefCell::try_borrow`](struct.RefCell.html#method.try_borrow).
+#[unstable(feature = "try_borrow", issue = "35070")]
+pub struct BorrowError<'a, T: 'a + ?Sized> {
+    marker: PhantomData<&'a RefCell<T>>,
+}
+
+impl<'a, T: ?Sized> BorrowError<'a, T> {
+    #[unstable(feature = "try_borrow_error_internals",
+               reason = "available through Error trait, and this method should \
+                         not be exposed publicly",
+               issue = "0")]
+    #[doc(hidden)]
+    pub fn __description(&self) -> &str {
```

Comment:
Despite this being how `std::num` does it in a few locations, that's just because the variants in core aren't available in std. For something like these errors it should be fine to just put this error string directly into std, no need to worry about reducing the duplication of this string literal!